src/io/tradetrust/certificate-of-origin/1.0/schema.test.ts

```diff
@@ -1,172 +1,76 @@
-          "message": "should have required property 'supplyChainConsignment'",
-          "params": Object {
-            "missingProperty": "supplyChainConsignment",
-          },
-          "schemaPath": "#/required",
-        },
-      ]
-    `);
+describe("certificate of origin schema", () => {
+  it("should work with valid certificate of origin", async () => {
+    const mergedDocument = {
+      ...CertificateOfOriginSchema,
+      ...CertificateOfOriginSampleData
+    } as JsonLdDocument;
+    expect(await expandDocument(mergedDocument)).toBeTruthy();
   });
 
-  it("should fail when document does not have exporter", () => {
-    const document = omit(cloneDeep(sampleDocument), "supplyChainConsignment.exporter");
+  it("should throw error when unknown context or properties found", async () => {
+    const modifiedData = {
+      ...CertificateOfOriginSampleData,
+      invalidCOOProperties: "Random String"
+    };
 
-    expect(validator(document)).toBe(false);
-    expect(validator.errors).toMatchInlineSnapshot(`
-      Array [
-        Object {
-          "dataPath": ".supplyChainConsignment",
-          "keyword": "required",
-          "message": "should have required property 'exporter'",
-          "params": Object {
-            "missingProperty": "exporter",
-          },
-          "schemaPath": "#/properties/supplyChainConsignment/required",
-        },
-      ]
-    `);
-  });
+    const mergedDocument = {
+      ...CertificateOfOriginSchema,
+      ...modifiedData
+    } as JsonLdDocument;
 
-  it("should fail when document does not have exporter's iD, name, postalAddress", () => {
-    const document = {
-      ...sampleDocument,
-      supplyChainConsignment: {
-        ...sampleDocument.supplyChainConsignment,
-        exporter: {}
-      }
-    };
-    expect(validator(document)).toBe(false);
-    expect(validator.errors).toMatchInlineSnapshot(`
-      Array [
-        Object {
-          "dataPath": ".supplyChainConsignment.exporter",
-          "keyword": "required",
-          "message": "should have required property 'iD'",
-          "params": Object {
-            "missingProperty": "iD",
-          },
-          "schemaPath": "#/allOf/1/required",
-        },
-        Object {
-          "dataPath": ".supplyChainConsignment.exporter",
-          "keyword": "required",
-          "message": "should have required property 'name'",
-          "params": Object {
-            "missingProperty": "name",
-          },
-          "schemaPath": "#/allOf/1/required",
-        },
-        Object {
-          "dataPath": ".supplyChainConsignment.exporter",
-          "keyword": "required",
-          "message": "should have required property 'postalAddress'",
-          "params": Object {
-            "missingProperty": "postalAddress",
-          },
-          "schemaPath": "#/allOf/1/required",
-        },
-      ]
-    `);
+    return expandDocument(mergedDocument).catch(e => {
+      expect(e.message).toMatchInlineSnapshot(
```

Comment:
https://jestjs.io/docs/tutorial-async#rejects

Reply:
done. updated test to read .rejects and expect throwError